darts/models/forecasting/prophet.py

```diff
@@ -3,34 +3,57 @@
 ----------------
 """
 
-from typing import Optional
+from typing import Optional, Union, List
 import logging
+import numpy as np
 
 from darts.timeseries import TimeSeries
-from darts.models.forecasting.forecasting_model import ForecastingModel
+from darts.models.forecasting.forecasting_model import DualCovariatesForecastingModel
 import pandas as pd
-from darts.logging import get_logger, execute_and_suppress_output
+from darts.logging import get_logger, execute_and_suppress_output, raise_if
 import prophet
 
 
 logger = get_logger(__name__)
 logger.level = logging.WARNING  # set to warning to suppress prophet logs
 
 
-class Prophet(ForecastingModel):
+class Prophet(DualCovariatesForecastingModel):
```

Comment:
Technically I think that this should be a `FutureCovariatesForecastingModel`, because at prediction time it requires only the future values of the future covariates (not its historic values).
However I think we actually should also change this for a few other models which are currently inheriting from `DualCovariatesForecastingModel`; so we can perhaps leave that for another time.